Re-land Create SPI for Enhanced Security

darryl-apple opened the pull request:
#### 347f8728cdd8157b9ac6cd45390e38526c31fcf3
<pre>
Re-land Create SPI for Enhanced Security
<a href="https://bugs.webkit.org/show_bug.cgi?id=299285">https://bugs.webkit.org/show_bug.cgi?id=299285</a>
<a href="https://rdar.apple.com/161090983">rdar://161090983</a>

Reviewed by Per Arne Vollan.

Create SPI for EnhancedSecurity

Tests: security/cocoa/check-enhanced-security-disabled.html
       security/cocoa/check-enhanced-security-enabled.html
       security/cocoa/enhanced-security/check-enhanced-security-enabled.html
       security/cocoa/lockdown-mode/check-lockdown-mode-enabled.html
       Tools/TestWebKitAPI/SourcesCocoa.txt
       Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj
       Tools/TestWebKitAPI/Tests/WebKitCocoa/EnhancedSecurity.mm
* LayoutTests/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/security/cocoa/check-enhanced-security-disabled-expected.txt: Added.
* LayoutTests/security/cocoa/check-enhanced-security-disabled.html: Added.
* LayoutTests/security/cocoa/check-enhanced-security-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/check-enhanced-security-enabled.html: Added.
* LayoutTests/security/cocoa/enhanced-security/check-enhanced-security-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/enhanced-security/check-enhanced-security-enabled.html: Added.
* LayoutTests/security/cocoa/lockdown-mode/check-lockdown-mode-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/lockdown-mode/check-lockdown-mode-enabled.html: Added.
* Source/WebCore/page/Page.h:
(WebCore::Page::setWebContentProcessVariant):
(WebCore::Page::webContentProcessVariant const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::webContentProcessVariant const):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.h:
(WebKit::XPCServiceInitializer):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
(WebKit::XPCServiceInitializerDelegate::getExtraInitializationData):
(WebKit::setJSCOptions):
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::enhancedSecurityEnabled const):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _isEnhancedSecurityEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferences.mm:
(-[WKWebpagePreferences _setEnhancedSecurityEnabled:]):
(-[WKWebpagePreferences _enhancedSecurityEnabled]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferencesPrivate.h:
* Source/WebKit/UIProcess/BrowsingContextGroup.cpp:
(WebKit::BrowsingContextGroup::sharedProcessForSite):
* Source/WebKit/UIProcess/BrowsingContextGroup.h:
* Source/WebKit/UIProcess/SuspendedPageProxy.cpp:
(WebKit::SuspendedPageProxy::findReusableSuspendedPageProcess):
* Source/WebKit/UIProcess/SuspendedPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::launchProcess):
(WebKit::WebPageProxy::receivedNavigationActionPolicyDecision):
(WebKit::WebPageProxy::triggerBrowsingContextGroupSwitchForNavigation):
(WebKit::WebPageProxy::isEnhancedSecurityEnabled):
(WebKit::WebPageProxy::shouldEnableEnhancedSecurity const):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessCache.cpp:
(WebKit::WebProcessCache::takeProcess):
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::establishRemoteWorkerContextConnectionToNetworkProcess):
(WebKit::WebProcessPool::createNewWebProcess):
(WebKit::WebProcessPool::tryTakePrewarmedProcess):
(WebKit::WebProcessPool::prewarmProcess):
(WebKit::WebProcessPool::processForSite):
(WebKit::WebProcessPool::createWebPage):
(WebKit::WebProcessPool::processForNavigation):
(WebKit::WebProcessPool::prepareProcessForNavigation):
(WebKit::WebProcessPool::processForNavigationInternal):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::create):
(WebKit::WebProcessProxy::createForRemoteWorkers):
(WebKit::WebProcessProxy::WebProcessProxy):
(WebKit::WebProcessProxy::getLaunchOptions):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_toolbarsAreVisible):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeProcess):
(WebKit::WebProcess::isEnhancedSecurityEnabled):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Tools/MiniBrowser/mac/SettingsController.h:
* Tools/MiniBrowser/mac/SettingsController.m:
(-[SettingsController _populateMenu:]):
(-[SettingsController validateMenuItem:]):
(-[SettingsController toggleEnhancedSecurityEnabled:]):
(-[SettingsController enhancedSecurityEnabled]):
* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController didChangeSettings]):
* Tools/TestRunnerShared/TestFeatures.cpp:
(WTR::shouldEnableEnhancedSecurity):
(WTR::hardcodedFeaturesBasedOnPathForTest):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EnhancedSecurity.mm: Added.
(TestWebKitAPI::isEnhancedSecurityEnabled):
(TestWebKitAPI::isJITEnabled):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityEnablesTrue)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityEnableFalse)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityDisablesJIT)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityNavigationStaysEnabledAfterNavigation)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecurity)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySamePage)):
(TestWebKitAPI::psonProcessPoolConfiguration):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySharedProcessPool)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySharedProcessPoolReverse)):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::enhancedSecurityEnabled const):
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::configureWebpagePreferences):

Canonical link: <a href="https://commits.webkit.org/300457@main">https://commits.webkit.org/300457@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6ac8c240c0f5d0aefc5e1c799baacc54301f2f62

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122493 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42201 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32886 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129096 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/74592 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b3eb6dec-7307-4a5f-afd0-4efdd8be80fd) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/124369 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42919 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50794 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93122 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61846 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/39832b65-613f-4645-9ac6-0931872f2bda) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125445 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34235 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109679 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73764 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fc9f9605-fcf0-4dd5-bcc5-a73beab77924) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33223 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27836 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72584 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103882 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28047 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131825 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49434 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37624 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101660 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49808 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105900 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101528 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25795 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46890 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25037 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/46203 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49292 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55041 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48759 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52109 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50441 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WebCore/testing/Internals.h:
```diff
@@ -1035,6 +1035,8 @@ class Internals final
 
     bool isHardwareVP9DecoderExpected();
 
+    bool enhancedSecurityEnabled() const;
```
As I mentioned in another PR, while I think having the term "enhanced security" in Apple's WebKit API/SPI is fine, I really don't think it belongs in the internals as it is not a term that is meaningful or has standards associated with it.

darryl-apple replied:
Hey - Thanks for the feedback! We'll look at breaking it down into more specifics as the feature develops

weinig replied:
Can you go into more detail about why WebCore internals need to know about this state at the moment?

gavin-apple replied:
This is purely to expose to LayoutTests when we are/aren't running in a given configuration. We actually want the same thing for LockdownMode too. 

How about an alternate proposal of `DOMString webContentProcessVariant()` which returns 'standard', 'lockdown', 'security', etc.? That way, we can have a generic routine which can be re-used throughout the various testing scenarios we would like to cover.

weinig replied:
Those seem like things that are probably better tested through API tests.

pvollan commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -6560,6 +6560,25 @@ bool Internals::audioSessionActive() const
     return false;
 }
 
+String Internals::webContentProcessVariant() const
+{
+    auto* document = contextDocument();
+    if (!document)
+        return { };
+    auto* page = document->page();
+    if (!page)
+        return { };
+
+    switch (page->webContentProcessVariant()) {
+    case WebContentProcessVariant::Security:
+        return "security"_s;
+    case WebContentProcessVariant::Lockdown:
+        return "lockdown"_s;
+    default:
+        return "standard"_s;
+    }
+}
```
I wonder if we should use entitlements checks instead to determine the WebContent process variant. What do you think?

gavin-apple replied:
I agree that we should do this and is the intent. But this PR doesn’t have the code to spin the new process type up - that’s in the follow-up. We didn’t want to merge the 2 PRs and it be too large to review. The hope was that this small amount of glue code would be enough to get an initial set of tests passing then in the follow-up we’d replace it with a process check using entitlement/process name or similar.

pvollan replied:
Sounds good, thanks! Should we file a bug, and add a comment about this? I think we can move forward with the current approach, if Sam does not have concerns with the Internals changes.

pvollan replied:
cc @weinig

weinig replied:
WebCore, and therefore this Internals class, really should not know anything about web process, as that is a WebKit concept.

I do think this would be better as an API test.

pvollan replied:
That's a good point @weinig! @darryl-apple, would you have time to create API tests for this instead?

pvollan commented on Source/WebCore/page/Page.h:
```diff
@@ -1380,6 +1386,9 @@ class Page : public RefCountedAndCanMakeWeakPtr<Page>, public Supplementable<Pag
     void setHardwareKeyboardAttached(bool attached) { m_hardwareKeyboardAttached = attached; }
     bool hardwareKeyboardAttached() const { return m_hardwareKeyboardAttached; }
 
+    void setWebContentProcessVariant(WebContentProcessVariant value) { m_webContentProcessVariant = value; };
```
Nit: `value` -> `variant`

darryl-apple replied:
Changed

pvollan commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -12227,6 +12229,12 @@ void WebPageProxy::isJITEnabled(CompletionHandler<void(bool)>&& completionHandle
 
+void WebPageProxy::isEnhancedSecurityEnabled(CompletionHandler<void(bool)>&& completionHandler)
+{
+    launchInitialProcessIfNecessary();
```
Should we just call the completion handler with `false` if a WebContent process hasn't been launched yet?

darryl-apple replied:
Yes that seems sensible. Changed

pvollan commented on Source/WebKit/WebProcess/WebProcess.messages.in:
```diff
@@ -145,6 +145,7 @@ messages -> WebProcess : AuxiliaryProcess WantsAsyncDispatchMessage {
 #endif
 
     IsJITEnabled() -> (bool enabled)
+    IsEnhancedSecurityEnabled() -> (bool enabled)
```
We may replace this with an entitlement check at a later point.

darryl-apple committed:
Re-land Create SPI for Enhanced Security
https://bugs.webkit.org/show_bug.cgi?id=299285
rdar://161090983

Reviewed by Per Arne Vollan.

Create SPI for EnhancedSecurity

Tests: security/cocoa/check-enhanced-security-disabled.html
       security/cocoa/check-enhanced-security-enabled.html
       security/cocoa/enhanced-security/check-enhanced-security-enabled.html
       security/cocoa/lockdown-mode/check-lockdown-mode-enabled.html
       Tools/TestWebKitAPI/SourcesCocoa.txt
       Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj
       Tools/TestWebKitAPI/Tests/WebKitCocoa/EnhancedSecurity.mm
* LayoutTests/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/security/cocoa/check-enhanced-security-disabled-expected.txt: Added.
* LayoutTests/security/cocoa/check-enhanced-security-disabled.html: Added.
* LayoutTests/security/cocoa/check-enhanced-security-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/check-enhanced-security-enabled.html: Added.
* LayoutTests/security/cocoa/enhanced-security/check-enhanced-security-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/enhanced-security/check-enhanced-security-enabled.html: Added.
* LayoutTests/security/cocoa/lockdown-mode/check-lockdown-mode-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/lockdown-mode/check-lockdown-mode-enabled.html: Added.
* Source/WebCore/page/Page.h:
(WebCore::Page::setWebContentProcessVariant):
(WebCore::Page::webContentProcessVariant const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::webContentProcessVariant const):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.h:
(WebKit::XPCServiceInitializer):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
(WebKit::XPCServiceInitializerDelegate::getExtraInitializationData):
(WebKit::setJSCOptions):
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::enhancedSecurityEnabled const):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _isEnhancedSecurityEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferences.mm:
(-[WKWebpagePreferences _setEnhancedSecurityEnabled:]):
(-[WKWebpagePreferences _enhancedSecurityEnabled]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferencesPrivate.h:
* Source/WebKit/UIProcess/BrowsingContextGroup.cpp:
(WebKit::BrowsingContextGroup::sharedProcessForSite):
* Source/WebKit/UIProcess/BrowsingContextGroup.h:
* Source/WebKit/UIProcess/SuspendedPageProxy.cpp:
(WebKit::SuspendedPageProxy::findReusableSuspendedPageProcess):
* Source/WebKit/UIProcess/SuspendedPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::launchProcess):
(WebKit::WebPageProxy::receivedNavigationActionPolicyDecision):
(WebKit::WebPageProxy::triggerBrowsingContextGroupSwitchForNavigation):
(WebKit::WebPageProxy::isEnhancedSecurityEnabled):
(WebKit::WebPageProxy::shouldEnableEnhancedSecurity const):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessCache.cpp:
(WebKit::WebProcessCache::takeProcess):
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::establishRemoteWorkerContextConnectionToNetworkProcess):
(WebKit::WebProcessPool::createNewWebProcess):
(WebKit::WebProcessPool::tryTakePrewarmedProcess):
(WebKit::WebProcessPool::prewarmProcess):
(WebKit::WebProcessPool::processForSite):
(WebKit::WebProcessPool::createWebPage):
(WebKit::WebProcessPool::processForNavigation):
(WebKit::WebProcessPool::prepareProcessForNavigation):
(WebKit::WebProcessPool::processForNavigationInternal):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::create):
(WebKit::WebProcessProxy::createForRemoteWorkers):
(WebKit::WebProcessProxy::WebProcessProxy):
(WebKit::WebProcessProxy::getLaunchOptions):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_toolbarsAreVisible):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeProcess):
(WebKit::WebProcess::isEnhancedSecurityEnabled):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Tools/MiniBrowser/mac/SettingsController.h:
* Tools/MiniBrowser/mac/SettingsController.m:
(-[SettingsController _populateMenu:]):
(-[SettingsController validateMenuItem:]):
(-[SettingsController toggleEnhancedSecurityEnabled:]):
(-[SettingsController enhancedSecurityEnabled]):
* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController didChangeSettings]):
* Tools/TestRunnerShared/TestFeatures.cpp:
(WTR::shouldEnableEnhancedSecurity):
(WTR::hardcodedFeaturesBasedOnPathForTest):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EnhancedSecurity.mm: Added.
(TestWebKitAPI::isEnhancedSecurityEnabled):
(TestWebKitAPI::isJITEnabled):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityEnablesTrue)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityEnableFalse)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityDisablesJIT)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityNavigationStaysEnabledAfterNavigation)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecurity)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySamePage)):
(TestWebKitAPI::psonProcessPoolConfiguration):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySharedProcessPool)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySharedProcessPoolReverse)):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::enhancedSecurityEnabled const):
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::configureWebpagePreferences):

Canonical link: https://commits.webkit.org/300457@main
